web/app/js/components/Tap.jsx

```diff
@@ -0,0 +1,155 @@
+import _ from 'lodash';
+import ErrorBanner from './ErrorBanner.jsx';
+import PageHeader from './PageHeader.jsx';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { withContext } from './util/AppContext.jsx';
+import { Button, Form, Icon, Input } from 'antd';
+import './../../css/tap.css';
+
+class Tap extends React.Component {
+  static propTypes = {
+    pathPrefix: PropTypes.string.isRequired
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      errors: "",
+      resource: "",
+      namespace: "",
+      messages: [],
+      maxLinesToDisplay: 40,
+      webSocketRequestSent: false
+    };
+  }
+
+  componentWillUnmount() {
+    this.ws.close(1000);
+    this.stopTapStreaming();
+    this.stopServerPolling();
+  }
+
+  onWebsocketOpen = () => {
```

Comment:
`onWebsocketOpen` is defined twice now -- once here and once on line 64. I think you want to delete the other one and keep this one, since it resets `errors` in the state.

Reply:
Err, looking a bit more closely now, this version of `onWebsocketOpen` references `this.state.query`, but that's not being set anywhere. So I think you'll need some combination of this method and the other method.

Reply:
ahhhhh whoops, bad merge thanks!